packages/webapp/src/pages/Connection/Show.tsx

```diff
@@ -60,8 +60,9 @@ export default function ShowIntegration() {
 
         const getConnections = async () => {
             const res = await getConnectionDetailsAPI(connectionId, providerConfigKey, false);
-
-            if (res?.status === 200) {
+            if (res?.status === 400) {
+                navigate('/404');
```

Comment:
you don't want to redirect to a 404, you want to keep the URL so the user can know that the ressource they accessed is 404. 
To achieve this, you can remove the route and instead store a state to do something like this in the component
```tsx
if (is404) {
  return <NotFound />
}
```

Reply:
Implemented this